amqp.go

```diff
@@ -107,6 +116,19 @@ func (amqp *AMQP) Publish(options PublishOptions) error {
 		publishing.DeliveryMode = amqpDriver.Persistent
 	}
 
+	// Well, I guess 1970-01-01 isn't allowed now.
+	if options.Timestamp != 0 {
```

Comment:
I'm curious about the issue you saw when an explicit timestamp of 0 was provided. I removed the conditional and passed through a 0 timestamp without issue.

Reply:
In fairness I'm no go developer and may have misunderstood this. But if I read the docs right a struct with no value will get a default value, which for this would be 0 - and I didn't want everyone's messages sent with this to start having a timestamp set in instances where the author didn't intend it

Reply:
Ok, I get it. By the comment, I understood it as some error was encountered when specifying a value of zero. 

By not setting the `publishing.Timestamp`, does the driver default the timestamp to the current time? It may be better to set the publish timestamp to the current time rather than omit.

Reply:
Followup...ran tests locally. If the `publishing.Timestamp` is not provided, there is simply no `timestamp` property provided with the message. So basically, if you want to have timestamps on your messages, you must provide a non-zero value. I'm good with that.